src/snowflake/connector/connection.py

```diff
@@ -475,8 +488,16 @@ def login_timeout(self) -> int | None:
         return int(self._login_timeout) if self._login_timeout is not None else None
 
     @property
-    def network_timeout(self) -> int | None:
-        return int(self._network_timeout) if self._network_timeout is not None else None
+    def request_timeout(self) -> int | None:
+        return int(self._request_timeout) if self._request_timeout is not None else None
+
+    @property
+    def socket_timeout(self) -> int | None:
+        return int(self._socket_timeout) if self._socket_timeout is not None else None
+
+    @property
+    def backoff_generator(self) -> Iterator:
```

Comment:
let's make it private if this is internally used only
```suggestion
    def _backoff_generator(self) -> Iterator:
```